Codesandbox examples for charting demos

theiliad opened the pull request:
- Automatic demo generation using the Carbon Charts storybook examples
- Addition of CodeSandbox links to every demo
- Cleaning up the dataviz demo pages

Closes #1271

vpicone commented on src/components/ComponentDemo/ComponentDemo.js:
```diff
@@ -55,7 +58,7 @@ const ComponentDemo = ({
   // Tests/cleanup context
   // handle true/false better
-
+  console.log('theme', theme, theme.indexOf('g100') > -1);
```
Is this left over?

theiliad replied:
yes, actually I'll remove all changes to `ComponentDemo` since they're from an older approach and not needed

vpicone commented on src/components/ComponentDemo/ComponentDemo.js:
```diff
@@ -91,6 +99,7 @@ const ComponentDemo = ({
             )}
             <Code links={links} code={code} src={src}>
               <LiveEditor
+                disabled
```
We don't want to disable the live editor?

vpicone replied:
You could make this a prop for the Component if you want.

vpicone commented on src/data/charts/index.js:
```diff
@@ -1,7 +1,8 @@
 
+export * from '@carbon/charts/demo/data';
+
+console.log('de', de);
```
console log

vpicone commented on src/pages/data-visualization/basic-charts/index.mdx:
```diff
@@ -9,6 +9,8 @@ Basic charts offer a way to visualize data sets in an intuitive, easy to underst
 
 </PageDescription>
 
+import { CodeSnippet } from 'carbon-components-react';
```
extra import?

vpicone commented:
Sidenote, @jnm2377 is doing a lot of work on the Component demo now. We're making some pretty big changes so you might want to hold off on this. We can't be responsible for any breaks to your usage of the component.

theiliad commented:
> Sidenote, @jnm2377 is doing a lot of work on the Component demo now. We're making some pretty big changes so you might want to hold off on this. We can't be responsible for any breaks to your usage of the component.

Yeah the `ComponentDemo` change were part of an older approach I was taking that came in accidentally with this PR. I've removed all of those changes

theiliad commented:
ready for reviews here

emyarod commented on src/pages/data-visualization/chart-types/index.mdx:
```diff
@@ -16,8 +16,8 @@ Start by identifying the purpose of the visualization and choose a chart type ap
-    title="Bar chart"
-    href="/data-visualization/basic-charts#bar-chart"
+    title="Simple bar (vertical)"
+    href="/data-visualization/basic-charts#simple-bar-(discrete)"
```
most of the OverviewCards link to `h2`s but others link to the chart demo wrappers directly. should they all follow the same pattern (or a similar pattern, linking to a description first rather than the chart demo itself)? (cc @janchild for content review)

tw15egan commented:
Noticed the Radar link hover doesn't seem to work? Rest work ok 👍 

![data-viz](https://user-images.githubusercontent.com/11928039/84190487-de799300-aa4b-11ea-8242-c567ab6d9c6a.gif)

theiliad commented:
That might be an issue in the component, not sure if it’s related to this PR

On Tue, Jun 9, 2020 at 3:23 PM TJ Egan <notifications@github.com> wrote:

> Noticed the Radar link hover doesn't seem to work? Rest work ok 👍
>
> [image: data-viz]
> <https://user-images.githubusercontent.com/11928039/84190487-de799300-aa4b-11ea-8242-c567ab6d9c6a.gif>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/carbon-design-system/carbon-website/pull/987#issuecomment-641520988>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADSLT3BLU3CNX5QJMZRYKCDRV2D2NANCNFSM4LCSOBMQ>
> .
>

tw15egan commented:
@theiliad seems like the image has a background, the rest have a transparent background. I can create a separate issue for that since it's unrelated to this change.

vpicone commented on src/pages/data-visualization/chart-types/index.mdx:
```diff
@@ -127,32 +126,32 @@ Start by identifying the purpose of the visualization and choose a chart type ap
     >
 
-![](images/treemap.svg)
+![](images/areastack.svg)
```
Wee need to have alt tags for these

theiliad commented:
Hi, any updates around timeline? Are we able to merge today?

theiliad commented:
bumping this again

mjabbink commented:
tags are broken on “Chart types” page


<img width="855" alt="Screen Shot 2020-06-16 at 1 25 26 PM" src="https://user-images.githubusercontent.com/32881239/84812993-e2219280-afd4-11ea-978b-7882f788b69b.png">

theiliad commented:
> tags are broken on “Chart types” page
> 
> <img alt="Screen Shot 2020-06-16 at 1 25 26 PM" width="855" src="https://user-images.githubusercontent.com/32881239/84812993-e2219280-afd4-11ea-978b-7882f788b69b.png">

that's irrelevant to this PR, that's currently the behaviour you'll see on the site

mjabbink commented:
Column widths are not right for both demo container and for body copy.
— duo too narrow
— body too wide


<img width="1102" alt="Screen Shot 2020-06-16 at 1 57 41 PM" src="https://user-images.githubusercontent.com/32881239/84816064-8efe0e80-afd9-11ea-82a2-3faba703336e.png">

designertyler commented:
The headers are not getting links when hovered. 
![Jun-16-2020 14-19-31](https://user-images.githubusercontent.com/19270502/84818689-80fdbd00-afdc-11ea-86f2-6dcd2e729e88.gif)

theiliad commented:
> The headers are not getting links when hovered.
> ![Jun-16-2020 14-19-31](https://user-images.githubusercontent.com/19270502/84818689-80fdbd00-afdc-11ea-86f2-6dcd2e729e88.gif)

addressed


> Column widths are not right for both demo container and for body copy.
> — duo too narrow
> — body too wide
> 
> <img alt="Screen Shot 2020-06-16 at 1 57 41 PM" width="1102" src="https://user-images.githubusercontent.com/32881239/84816064-8efe0e80-afd9-11ea-82a2-3faba703336e.png">

addressed

> tags are broken on “Chart types” page
> 
> <img alt="Screen Shot 2020-06-16 at 1 25 26 PM" width="855" src="https://user-images.githubusercontent.com/32881239/84812993-e2219280-afd4-11ea-978b-7882f788b69b.png">

addressed

theiliad commented:
> tags are broken on “Chart types” page
> <img alt="Screen Shot 2020-06-16 at 1 25 26 PM" width="855" src="https://user-images.githubusercontent.com/32881239/84812993-e2219280-afd4-11ea-978b-7882f788b69b.png">

actually I'll undo this change since #1305 will push the fix

mjabbink commented:
All looking good but I have one question regarding certain vertical lines aligning to the week dates in time series. The lines do not align to the days/dates that are visible? This seems like a mistake.

<img width="825" alt="Screen Shot 2020-06-22 at 12 04 16 PM" src="https://user-images.githubusercontent.com/32881239/85315564-e7715800-b480-11ea-8863-40535c7661d2.png">

——————

<img width="798" alt="Screen Shot 2020-06-22 at 12 04 41 PM" src="https://user-images.githubusercontent.com/32881239/85315568-e9d3b200-b480-11ea-82f5-93a6861f08b8.png">

——————

<img width="757" alt="Screen Shot 2020-06-22 at 12 04 59 PM" src="https://user-images.githubusercontent.com/32881239/85315591-f5bf7400-b480-11ea-9bfa-b069c9ccbea2.png">

——————

<img width="769" alt="Screen Shot 2020-06-22 at 12 05 14 PM" src="https://user-images.githubusercontent.com/32881239/85315603-fbb55500-b480-11ea-9379-6921b9f1ecbd.png">

theiliad commented:
> All looking good but I have one question regarding certain vertical grid lines aligning to the week dates in time series. The lines do not align to the days/dates that are visible? This seems like a mistake. @shixiedesign @jeanservaas @cameroncalder
> 
> <img alt="Screen Shot 2020-06-22 at 12 04 16 PM" width="825" src="https://user-images.githubusercontent.com/32881239/85315564-e7715800-b480-11ea-8863-40535c7661d2.png">
> 
> ——————
> 
> <img alt="Screen Shot 2020-06-22 at 12 04 41 PM" width="798" src="https://user-images.githubusercontent.com/32881239/85315568-e9d3b200-b480-11ea-82f5-93a6861f08b8.png">
> 
> ——————
> 
> <img alt="Screen Shot 2020-06-22 at 12 04 59 PM" width="757" src="https://user-images.githubusercontent.com/32881239/85315591-f5bf7400-b480-11ea-9bfa-b069c9ccbea2.png">
> 
> ——————
> 
> <img alt="Screen Shot 2020-06-22 at 12 05 14 PM" width="769" src="https://user-images.githubusercontent.com/32881239/85315603-fbb55500-b480-11ea-9379-6921b9f1ecbd.png">

again, separate issue irrelevant to this PR.

theiliad commented:
the build has not started after 45 minutes which is why github is showing the red X beside vercel, @vpicone could you restart it please?

mjabbink commented:
aren’t those the sandbox examples? If this is about just the layout of the page then this looks good to go.

theiliad commented:
> aren’t those the sandbox examples? If this is about just the layout of the page then this looks good to go.

this PR's job is to add our existing sandbox examples to the carbon website. what you're mentioning would be a bug (if broken) in the actual Carbon Charts library. Also, labels were refactored by accurat, so this is either something that they missed or didn't start with as a requirement

theiliad commented:
> Resource card needs to be under the anchor links

done @mjabbink

mjabbink commented:
Don’t see it reflected in the preview but assume it follows all the other pages with resource card placed under the intro and anchor links. Good to merge otherwise.

theiliad commented:
> Don’t see it reflected in the preview but assume it follows all the other pages with resource card placed under the intro and anchor links. Good to merge otherwise.

yes, the preview is still building, probably another 10-15minutes but yes I've moved the resources card below anchor links

mjabbink commented:
@theiliad See it now and good to go.

theiliad commented:
> @theiliad See it now and good to go.

great thanks Mike

vpicone commented on src/components/ComponentDemo/Code/CodeBar.js:
```diff
@@ -17,21 +17,24 @@ const StorybookLink = ({ framework, url }) => (
-        <a target="_blank" rel="noopener noreferrer" href={sandboxUrl}>
-          CodeSandbox <Launch16 />
-        </a>
+        {showSandboxURL && (
```
Love the re-use of the code bar. I'm confused how `showSandboxURL` is meant to be used though. As is, this check would mean we have to add `showSandboxURL=true` to all of our deployed ComponentDemos (they're currently missing the sandbox link in build preview).

What if instead, `useCodesandbox` returned null when no code is provided. We could use `sandboxUrl &&` to determine whether to show the link or not.

```suggestion
        {sandboxUrl && (
```

theiliad replied:
good catch

vpicone commented on src/components/ComponentDemo/Code/CodeBar.js:
```diff
@@ -42,7 +45,9 @@ const CodeBar = ({ src, code, links }) => {
-      ) : (
+      )}
+
+      {!src && code && (
```
```suggestion
      {shouldShowCopyButton (
```

vpicone commented on src/components/ComponentDemo/Code/CodeBar.js:
```diff
@@ -17,21 +17,24 @@ const StorybookLink = ({ framework, url }) => (
+const CodeBar = ({ src, code, links, showSandboxURL }) => {
   const sandboxUrl = useCodesandbox(code);
   const storybookLinks = Object.entries(links);
 
```
**subjective alert** Do you mind moving this into a single, affirmative boolean. Just to clarify the intent a bit we can avoid chaining shortcuts. Not sure how you would have known this, it should be codified in an eslint rule.

```suggestion
const shouldShowCopyButton = !src && code;

```

theiliad commented:
> Some really minor things, great work re-purposing this component!

all addressed, I'd probably squash & merge since there are tons of commits

mjabbink commented:
👍
